### PR TITLE
Support addresses with scheme

### LINKED ISSUE #6: Would it be possible to support the http protocol?
```
❯ git get https://github.com/thechangelog/changelog.com
Cloning into '/Users/nathany/src/https/github.com/thechangelog/changelog.com'...
fatal: I don't handle protocol 'git@https'

❯ git get github.com/thechangelog/changelog.com        
fatal: repository 'github.com/thechangelog/changelog.com' does not exist
```

😦

### PR DESCRIPTION
**Pull request by homburg**
ie. https://github.com/some/thing.git
fixes #6 

@nathany Is this usable?

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -114,6 +114,51 @@ func runOrExit(cmd *exec.Cmd) *exitStatusError {
...
+
+func parseRepo(repo, host string) (string, []string) {
+	repo = strings.TrimSuffix(repo, ".git")
+
+	var repoParts []string
+	if strings.Count(repo, ":") == 1 {
+		/// if "@" in repo:
+		if strings.Contains(repo, "@") {
+			///	repo_parts = repo[repo.find("@")+1:].replace(":", "/").split("/")
+			repoParts = strings.Split(strings.Replace(repo[strings.Index(repo, "@")+1:len(repo)], ":", "/", -1), "/")
```
Comment (nathany) [inline main.go]: This function is a bit hard to follow.

File: main_test.go [hunk 2]
```diff
@@ -42,3 +41,61 @@ func TestDirIsEmpty(t *testing.T) {
...
+func TestRepoHasScheme(t *testing.T) {
+	testData := map[string]bool{
+		"git@github.com:some/thing.git":              false,
+		"http://what.dk/ever.git":                    true,
+		"https://github.com/something/different.git": true,
+	}
+
+	for addr, expected := range testData {
+		if hasScheme(addr) != expected {
+			t.Fatalf(`Expected scheme presence: %q for "%s", got %q.`, expected, addr, !expected)
```
Comment (nathany) [inline main_test.go]: For table driven tests like this, it may be more appropriate to use `t.Errorf` so all tests are exercised and reported on even if one of them fails.

**Review by nathany — COMMENTED** [review]
I haven't tested it out yet, but the general idea seems good to me. Supporting a scheme while falling back to SSH if no scheme is specified.

File: main_test.go (hunk 2 above)
Comment (homburg) [reply main_test.go]: Absolutely, thanks :+1:

**Comment by homburg** [comment]
Thank you for the feedback. I have split the parsers and tried to simplify the "ssh repo parser" with more intermediate variables and meaningful names.
